lib/rules/no-throw-literal.js

```diff
@@ -1,12 +1,32 @@
 /**
  * @fileoverview Rule to restrict what can be thrown as an exception.
  * @author Dieter Oberkofler
+ * @copyright 2015 Ian VanSchooten. All rights reserved.
  * @copyright 2015 Dieter Oberkofler. All rights reserved.
  */
 
 "use strict";
 
 //------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determine if a node is considered literal (including concatenation)
+ * @param  {ASTNode}  node  ASTNode to check for only literals
+ * @returns {Boolean}       True if considered literal, false otherwise
+ */
+function isLiteral(node) {
+    if (node.type === "Literal") {
+        return true;
+    }
+
+    if (node.type === "BinaryExpression") {
+        return true;
```

Comment:
"literal" is such as `true`, `0`, `100`, `"text"`, `null`, `{foo: 1}`, `[1, 2, 3]`, `function() {}`, ...
In short, it's the notation for a static constant value.

So, actually this rule name was wrong, but I believe people can guess the purpose of this rule from this name.  I guessed "I should throw an `Error` object".

I thought what my best effort is below.  I'm sure I cannot detect all patterns, but when that function returned `true`, the result of the expression should be not an error object.  (I did not run the code)

```js
function isNotErrorObject(node) {
    switch (node.type) {
        case "Identifier":
        case "CallExpression":
        case "NewExpression":
        case "MemberExpression":
        case "TaggedTemplateExpression":
        case "YieldExpression":
            return false; // possibly an error object.

        case "AssignmentExpression":
            return isNotErrorObject(node.right);

        case "SequenceExpression":
            var exprs = node.expressions;
            return exprs.length === 0 || isNotErrorObject(exprs[exprs.length - 1]);

        case "LogicalExpression"
            return isNotErrorObject(node.left) || isNotErrorObject(node.right);

        case "ConditionalExpression"
            return isNotErrorObject(node.consequent) || isNotErrorObject(node.alternate);

        default:
            return true;
    }
}
```